File: app/models/attendance.rb
Attendance = Struct.new(*%i[days count]) do
  def total
    days * count
  end

  def self.from_attendance_summary(participations_string)
    return [] unless participations_string
    participations_string.split(',').map do |participation_string|
      count, days = participation_string.split('x')
      new(days.to_f, count.to_i)
    end
  end
end

File: src/models/attendance.ts

export type Attendance = {
  days: number
  count: number
}

export function totalAttendance(attendance: Attendance) {
  return attendance.days * attendance.count
}

export function getAttendancesFromSummary(participationsString: string): Attendance[] {
  return participationsString.split(',').map(participationString => {
    const values = participationString.split('x')
    return {
      count: parseFloat(values[0]), days: parseFloat(values[1])
    }
  })
}

